server/src/main/java/org/elasticsearch/search/aggregations/bucket/terms/SignificanceLookup.java

```diff
@@ -214,7 +216,18 @@ public void close() {
      * Get the background frequency of a {@code long} term.
      */
     private long getBackgroundFrequency(long term) throws IOException {
-        return getBackgroundFrequency(context.buildQuery(new TermQueryBuilder(fieldType.name(), format.format(term).toString())));
+        return getBackgroundFrequency(context.buildQuery(makeBackgroundFrequencyQuery(format.format(term).toString())));
+    }
+
+    private QueryBuilder makeBackgroundFrequencyQuery(String value) {
+        QueryBuilder queryBuilder = new TermQueryBuilder(fieldType.name(), value);
+
+        var nestedParentField = context.nestedLookup().getNestedParent(fieldType.name());
+        if (nestedParentField != null) {
```

Comment:
Not sure if this is the best way to detect a nested field, but it worked well, and I didn't find any edge case

Reply:
I wonder if `buildQuery` should take the nested context into account? That sounds like a bigger change than I'd like to make to aggs at the moment.

Reply:
It has been some time since I worked on nested related functionality. Assuming that `fieldType.name()` returns a path, then  I think this is the right way to figure out the parent field.